### PR TITLE
fix(gatsby): render QoD overlay in its own root element to avoid hydration issues

### PR DESCRIPTION
**Pull request by KyleAMathews**
Clears way to enabling DEV_SSR for everyone.

### CODE REVIEW

File: packages/gatsby/cache-dir/app.js [hunk 1]
```diff
@@ -13,6 +13,8 @@ import asyncRequires from "$virtual/async-requires"
 // Generated during bootstrap
 import matchPaths from "$virtual/match-paths.json"
 
+import { LoadingIndicatorEventHandler } from "./loading-indicator"
```
Comment (wardpeet) [inline packages/gatsby/cache-dir/app.js:16]:
```suggestion
import { LoadingIndicatorEventHandler } from "./loading-indicator"
```

**Review by wardpeet — COMMENTED** [review]
Still need to test but added some small nits. Code looks good

**Review by pieh — APPROVED** [review]

**Comment by ascorbic** [comment]
This won't apply cleanly to the release branch, as you seem to have changed other things in there. Could you do a new PR against the release branch?

**Comment by KyleAMathews** [comment]
On it 👍

**Comment by ascorbic** [comment]
Published in `gatsby@2.32.7`

**Pull request merged**